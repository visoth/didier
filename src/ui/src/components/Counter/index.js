// @flow
import React from 'react'
import { injectIntl } from 'react-intl'
import type { Action } from '/routes/Counter/modules/counter.model'

type Props = {
  counter: number,
  intl: Object,
  actions: {
    increment: () => Action,
    doubleAsync: () => Action,
  }
}

export const Counter = (props: Props) => (
  <div>
    <h2>{props.intl.messages.counter_name}: {props.counter}</h2>
    <button className='btn btn-default' onClick={props.actions.increment}>
      {props.intl.messages.increment}
    </button>
    {' '}
    <button className='btn btn-default' onClick={props.actions.doubleAsync}>
      {props.intl.messages.double_async}
    </button>
  </div>
)

export default injectIntl(Counter)
