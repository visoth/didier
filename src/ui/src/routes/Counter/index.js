// @flow
import type { Store } from 'redux'

import { injectReducer } from '../../store/reducers'
import asyncComponent from '/routes/App/AsyncComponent'

export default (store: Store<*, *>) => asyncComponent(() => new Promise((resolve) => {
  /*  Webpack - use 'require.ensure' to create a split point
  and embed an async module loader (jsonp) when bundling   */
  require.ensure([], (require) => {
    /*  Webpack - use require callback to define
    dependencies for bundling   */
    const Counter = require('./containers/CounterContainer').default
    const reducer = require('./modules/counter.reducers').default

    /*  Add the reducer to the store on key 'counter'  */
    injectReducer(store, { key: 'counter', reducer })

    /*  Return getComponent   */
    resolve(Counter)
    /* Webpack named bundle   */
  }, 'counter')
}))
