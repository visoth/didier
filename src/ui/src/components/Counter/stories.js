import React from 'react'
import { storiesOf, action } from '@kadira/storybook'
import { withKnobs, text } from '@kadira/storybook-addon-knobs'
import { WithNotes } from '@kadira/storybook-addon-notes'

import Counter from './index'

storiesOf('Counter', module)
  .addDecorator(withKnobs)
  .add('counter at 2', () => (
    <WithNotes notes={`
      Lol: :D
    `}>
      <Counter
        counter={text('counter', '2')}
        actions={{
          increment: action('increment'),
          doubleAsync: action('doubleAsync')
        }}
      />
    </WithNotes>
  ))
