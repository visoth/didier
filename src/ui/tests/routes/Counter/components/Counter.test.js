/* eslint-disable */
import React from 'react'
import Counter from '/routes/Counter/components/Counter'

describe('Testing Counter component', () => {

    it('mounts Counter', () => {
        const wrapper = shallow(
            <Counter counter={2} actions={{
                increment: () => void(0),
                doubleAsync: () => void(0)
            }}/>
        );

        jespect(toJson(wrapper)).toMatchSnapshot();
    })

})
