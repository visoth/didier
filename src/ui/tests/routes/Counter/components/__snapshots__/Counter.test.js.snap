// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter component mounts Counter 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
    }
  }
>
  <h2>
    Counter: 
    2
  </h2>
  <button
    className="btn btn-default"
    onClick={[Function]}
  >
    Increment
  </button>
   
  <button
    className="btn btn-default"
    onClick={[Function]}
  >
    Double (Async)
  </button>
</div>
`;
